service/src/main/java/org/cbioportal/genome_nexus/service/cached/CachedEntrezGeneXrefFetcher.java

```diff
@@ -41,37 +41,40 @@ public GeneXref fetchAndCache(String accession, String geneSymbol)
         throws ResourceMappingException, HttpClientErrorException, ResourceAccessException
     {
         boolean saveValue = true;
-        GeneXref entrezGeneXref = null;
+        Optional<GeneXref> entrezGeneXrefOptional = null;
 
         try {
-            entrezGeneXref = this.repository.findOne(accession);
+            entrezGeneXrefOptional = this.repository.findById(accession);
         }
         catch (DataAccessResourceFailureException e) {
             LOG.warn("Failed to read from Mongo database - falling back on the external web service. " +
                 "Will not attempt to store variant in Mongo database.");
             saveValue = false;
         }
+         final boolean saveValueFinal = saveValue;
 
-        if (entrezGeneXref == null)
```

Comment:
Should we just use `if (entrezGeneXref.isPresent())` instead of `.map(x -> x)`? This would also eliminate the need of introducing a redundant `saveValueFinal` variable. Is there any advantages of using `orElseGet`?